### PR TITLE
Feature/schemas

### PR DESCRIPTION
**Pull request by gorold**
### Context
Updates pertaining to previous discussions with the Price Modelling team.

### Changes
- Slight changes to Constraint model
  - Updated to add strict inequalities
  - Added penalty field
- Added Schema and Header models
  - Schema represents the item IDs found in a DataBlock
  - Headers are the specifc item IDs
- Added ItemList and Item models
  - ItemList represents the cost sheet that is uploaded
  - Item contains the item ID, item name, and cost information.

### Considerations
- Users of the API will need to cross-reference item ids from Schema with item ids from ItemList in order to find the item names for a specific DataBlock.

### CODE REVIEW

File: secretsauce/apps/portal/models.py [hunk 1]
```diff
@@ -113,3 +120,37 @@ class ConstraintParameterRelationship(models.Model):
...
+class Item(models.Model):
+    """Item obtained from Cost Sheet"""
+    item_list = models.ForeignKey(
+        ItemList,
+        on_delete=models.CASCADE,
+        related_name='items',
+    )
+    name = models.CharField(max_length=200)
+    item_id = models.IntegerField()
+    cost = models.FloatField(blank=True)
```
Comment (eugenechia95) [inline secretsauce/apps/portal/models.py]: I find this whole idea of storing an array via two models kinda odd and redundant but I guess that is the way to go for what django fields provide. I know specifically for postgres, there is an Arrayfield for django but considering the extra work to set up locally and fixing on a single database, I think it is fine to not do it.

**Review by eugenechia95 — CHANGES_REQUESTED** [review]

File: secretsauce/apps/portal/models.py [hunk 2]
```diff
@@ -113,3 +120,37 @@ class ConstraintParameterRelationship(models.Model):
         on_delete=models.CASCADE,
     )
     coefficient = models.FloatField()
+
+class Schema(models.Model):
```
Comment (eugenechia95) [inline secretsauce/apps/portal/models.py:124]: 1. Maybe can change the name of `Schema` to `HeaderList`? I thought schema is quite a generic word referring to a typical database structure and does not provide much contextual meaning
2. I find the term `ItemList` a little odd semantically like when you use it as a view i.e. a list's list--- `class ItemListList(generics.ListCreateAPIView):`. Cant think of a better name too. If you cant think of one, I guess we can leave it the same

File: secretsauce/apps/portal/models.py (hunk 1 above)
Comment (eugenechia95) [reply secretsauce/apps/portal/models.py:156]: Also, so when creating constraints, we will use these items now? What if there is not cost sheet provided? Where would the items come from?

Comment (gorold) [reply secretsauce/apps/portal/models.py:156]: For constraint creation, decision variables should come from HeaderList, from the DataBlock. The item IDs from HeaderList can be cross-referenced with ItemList for their names and/or costs (items which appear on the ItemList and not HeaderList will also be available for cost data). If ItemList is not available, we will only present them with the item IDs from HeaderList.

File: secretsauce/apps/portal/models.py (hunk 2 above)
Comment (gorold) [reply secretsauce/apps/portal/models.py:124]: Sure, will make the changes.

**Review by eugenechia95 — APPROVED** [review]

**Pull request merged**